File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (vector<int> c : commands) {
        int start_idx = c[0];
        int end_idx = c[1];
        int k = c[2];
                vector<int> lst;
        for (int i=start_idx-1; i<end_idx; i++) {
            lst.push_back(array[i]);
        }
        
        sort(lst.begin(), lst.end());
        answer.push_back(lst[k-1]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int tmp = 0;
        
        for (int[] com : commands) {
            int i = com[0];
            int j = com[1];
            int k = com[2];
            
            List<Integer> newArray = new ArrayList<>();
            for (int x=i-1; x<j; x++) {
                newArray.add(array[x]);
            }
            
            Collections.sort(newArray);
            answer[tmp] = newArray.get(k-1);
            tmp++;
        }
        
        return answer;
    }
}